docs/how-to/h-manage-units.md

```diff
@@ -1,55 +1,102 @@
+be obtained using
 
-If you wish to run a command from the cluster, in order to (for example) list the current topics on the Kafka cluster, you can run:
 ```
 BOOTSTRAP_SERVERS=$(juju run kafka/leader get-admin-credentials | grep "bootstrap.servers" | cut -d "=" -f 2)
-juju ssh kafka/leader 'charmed-kafka.topics --bootstrap-server $BOOTSTRAP_SERVERS --list --command-config /var/snap/charmed-kafka/common/client.properties'
 ```
 
-Note that when no other application is related to Kafka, the cluster is secured-by-default and listeners are disabled, thus preventing any incoming connection. However, even for running the commands above, listeners must be enable. If there is no other application, deploy a `data-integrator` charm and relate it to Kafka, as outlined in the Relation section to enable listeners.
+Admin client authentication information are stored in the 
+`/var/snap/charmed-kafka/common/client.properties` file present on every Kafka
+broker. The content of the file can be accessed using 
 
-Available Kafka bin commands can be found with:
 ```
-snap info charmed-kafka --channel 3/edge
+juju ssh kafka/leader `cat /var/snap/charmed-kafka/common/client.properties`
 ```
 
-## Replication
+This file can be provided to the Kafka bin commands via the `--command-config`
+argument. Note that `client.properties` may also refer to other files (
+e.g. trustore and keystore for TLS-enabled connections) for which also those
+files need to be accessible and correctly specified. 
 
-### Scaling up
-The charm can be scaled up using `juju add-unit` command.
-```shell
-juju add-unit kafka
+Commands can also be run within a Kafka broker, since both the authentication 
+file (alongside with truststore if needed) and the Charmed Kafka snap are 
+already present. 
+
+#### Example (Listing topics)
+
+For instance, in order to list the current topics on the Kafka cluster, you can run:
+```
+juju ssh kafka/leader 'charmed-kafka.topics --bootstrap-server $BOOTSTRAP_SERVERS --list --command-config /var/snap/charmed-kafka/common/client.properties'
 ```
 
-To add a specific number of brokers, an extra argument is needed:
+### Juju External users
+
+For external users managed by the  [Data Integrator Charm](https://charmhub.io/data-integrator), 
+the endpoints and credentials can be fetched using the dedicated action
+
 ```shell
-juju add-unit kafka -n <num_brokers_to_add>
+juju run data-integrator/leader get-credentials --format yaml
+```
+
+The `client.properties` file can be generated by replacing relevant information in the 
+one available on the brokers, e.g. 
+
 ```
+BOOTSTRAP_SERVERS=$(juju run data-integrator/leader get-credentials --format yaml | yq .kafka.endpoints )
+USERNAME=$(juju run data-integrator/leader get-credentials --format yaml | yq .kafka.username )
+PASSWORD=$(juju run data-integrator/leader get-credentials --format yaml | yq .kafka.password )
+```
+
+## Replication and Scaling
```

Comment:
I think this should be moved up so it's the first section. It's the default instructions on how to scale with juju, in case that's all the user is looking for, and then more complex management details can follow for more advanced users. 
What do you think?